v2.0.6 and up: Way more than max_connections connections

I upgraded from v2.0.4 to v2.0.10, but when switching over the ProxySQL log was flooded with:

```
2020-03-27 10:12:40 mysql_connection.cpp:835:handler(): [ERROR] Failed to mysql_real_connect() on dbmaster:3306 , FD (Conn:1120 , MyDS:1120) , 1040: Too many connections.
2020-03-27 10:12:40 mysql_connection.cpp:835:handler(): [ERROR] Failed to mysql_real_connect() on dbmaster:3306 , FD (Conn:1120 , MyDS:1120) , 1040: Too many connections.
2020-03-27 10:12:40 mysql_connection.cpp:835:handler(): [ERROR] Failed to mysql_real_connect() on dbmaster:3306 , FD (Conn:1120 , MyDS:1120) , 1040: Too many connections.
2020-03-27 10:12:40 MySQL_HostGroups_Manager.cpp:905:connect_error(): [ERROR] Shunning server dbmaster:3306 with 5 errors/sec. Shunning for 10 seconds
```

I'm running ProxySQL v2.0.10 on Ubuntu 18.04 connecting to MariaDB 10.3.14.

After increasing several limits, timeouts, etc., I found out the following. On the ProxySQL side, in the (`runtime_`)`mysql_servers` table, `max_connections` is set to `200`. On the mysql side `max_connections` is `2000`. I had a few other processes connected directly to MySQL, and ProxySQL was using all of the remaining `2000` connections. Then I increased `max_connections` on the mysql side and within seconds ProxySQL was using 3000 more to saturate this. My upstream client uses a pool of size `100`, `mysql-max_connections` is set to `2048`, and with ProxySQL v2.0.4 there are only `58` open connections at this time, under the same load where ProxySQL v2.0.10 was using thousands.

Somehow there is a large difference in the two versions, and since this only happens in a production environment I am not so keen on debugging much further, but I am happy to provide more relevant info. To me this very much looks like a bug, but maybe  there is some config setting or change that I overlooked that could cause this?

--- a/lib/MySQL_Protocol.cpp
+++ b/lib/MySQL_Protocol.cpp
@@ -1549,6 +1549,14 @@ bool MySQL_Protocol::process_pkt_handshake_response(unsigned char *pkt, unsigned
 	}
 
 	capabilities     = CPY4(pkt);
+	// see bug #2916. If CLIENT_MULTI_STATEMENTS is set by the client
+	// we enforce setting CLIENT_MULTI_RESULTS, this is the proper and expected
+	// behavior (refer to 'https://dev.mysql.com/doc/c-api/8.0/en/c-api-multiple-queries.html').
+	// Don't enforcing this would cause a mismatch between client and backend
+	// connections flags.
+	if (capabilities & CLIENT_MULTI_STATEMENTS) {
+		capabilities |= CLIENT_MULTI_RESULTS;
+	}
 	(*myds)->myconn->options.client_flag = capabilities;
 	pkt     += sizeof(uint32_t);
 	max_pkt  = CPY4(pkt);

